docs/features/task-types.md

```diff
@@ -0,0 +1,90 @@
+Async Task Types in C#
```

Comment:
I think we're moving those documents to [csharplang](https://github.com/dotnet/csharplang/tree/master/spec).